plugin/trino-base-jdbc/src/test/java/io/trino/plugin/jdbc/CastDataTypeTest.java

```diff
@@ -0,0 +1,113 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.trino.plugin.jdbc;
+
+import com.google.common.collect.ImmutableList;
+import io.trino.testing.sql.SqlExecutor;
+import io.trino.testing.sql.TestTable;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Optional;
+import java.util.OptionalInt;
+import java.util.StringJoiner;
+
+import static com.google.common.base.Preconditions.checkArgument;
+import static com.google.common.base.Preconditions.checkState;
+import static java.util.Objects.requireNonNull;
+import static java.util.stream.Collectors.joining;
+
+public final class CastDataTypeTest
+{
+    private final List<TestCaseInput> testCaseInputs = new ArrayList<>();
+
+    private OptionalInt inputSize = OptionalInt.empty();
+    private TestTable testTable;
+
+    private CastDataTypeTest() {}
+
+    public static CastDataTypeTest create()
+    {
+        return new CastDataTypeTest();
+    }
+
+    public CastDataTypeTest addColumn(String columnName, String columnType, List<Object> inputValues)
+    {
+        if (inputSize.isEmpty()) {
+            inputSize = OptionalInt.of(inputValues.size());
+        }
+        checkArgument(inputSize.getAsInt() == inputValues.size(), "Expected input size: %s, but found: %s", inputSize.getAsInt(), inputValues.size());
```

Comment:
why is it needed?

Reply:
Just to ensure that caller should not provide the mismatched length for `inputValues`

Reply:
why caller should not provide the mismatched length for inputValues ?

Reply:
`inputValues`'s size indicates the total number of rows to be inserted into the table. If we have mismatched size for other columns, then It won't be valid.

I think I will `public static CastDataTypeTest create()` can be changed to `public static CastDataTypeTest create(int rowSize)` to accept the row size.

Reply:
I think it's better to haverow size implicit. 

Maybe `Suppliers.memoize` ?

Reply:
`Suppliers.memoize` will not verify. so no CR here.

Reply:
I liked an implicit approach, but explicit looks good too.